File: consumer-python/plugins/http_call_plugin.py
import os
import logging

import requests

from base_plugin import BasePlugin

log = logging.getLogger("plugin.http_call")

DEFAULT_URL = os.getenv("HTTP_PLUGIN_URL", "http://api:8000/webhook/mock")
TIMEOUT_SECONDS = 5


class HttpCallPlugin(BasePlugin):
    """Posts qualifying event payloads to a configurable HTTP endpoint."""

    @property
    def name(self) -> str:
        return "http_call_plugin"

    def handle(self, event: dict, settings: dict = None) -> None:
        target_url = (settings or {}).get("target_url", DEFAULT_URL)
        log.info("POST %s  event_type=%s", target_url, event.get("event_type"))
        resp = requests.post(target_url, json=event, timeout=TIMEOUT_SECONDS)
        resp.raise_for_status()
        log.info("HTTP %d from %s", resp.status_code, target_url)

File: consumer-node/src/plugins/httpCallPlugin.js
const axios = require("axios");
const BasePlugin = require("../basePlugin");
const pino = require("pino");

const log = pino({ name: "plugin.http_call" });

const DEFAULT_URL = process.env.HTTP_PLUGIN_URL || "http://api:8000/webhook/mock";
const TIMEOUT_MS = 5000;

class HttpCallPlugin extends BasePlugin {
  get name() {
    return "http_call_plugin_node";
  }

  async handle(event, settings) {
    const targetUrl = (settings && settings.target_url) || DEFAULT_URL;
    log.info({ url: targetUrl, event_type: event.event_type }, "POST event");

    const response = await axios.post(targetUrl, event, { timeout: TIMEOUT_MS });
    log.info({ status: response.status, url: targetUrl }, "HTTP response received");
  }
}

module.exports = HttpCallPlugin;
